### PR TITLE
ESP32: Move the BLE deinit after BLE connection is closed, post commissioning

### PR DESCRIPTION
**Pull request by dhrishi**
- Problem
The BLE deinit is done in kCommissioningComplete event. In some cases, it might happen that BLE connection close is not received from the commissioner but stil it tries to free BLE memory

- Solution
Perform deinit when BLE connection is closed with atleast one fabric being active

- Tests
Successful commissioning followed by reclaiming the BLE memory

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/platform/esp32/common/CommonDeviceCallbacks.cpp [hunk 1]
```diff
@@ -61,6 +62,43 @@ void CommonDeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, i
...
+                    nimble_port_deinit();
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+                    err = esp_nimble_hci_and_controller_deinit();
+#endif // ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+#endif // CONFIG_BT_NIMBLE_ENABLED
+
+#if CONFIG_IDF_TARGET_ESP32
+                    err += esp_bt_mem_release(ESP_BT_MODE_BTDM);
+#elif CONFIG_IDF_TARGET_ESP32C2 || CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32S3 || CONFIG_IDF_TARGET_ESP32H2
+            err += esp_bt_mem_release(ESP_BT_MODE_BLE);
```
Comment (chshu) [inline examples/platform/esp32/common/CommonDeviceCallbacks.cpp:86]:
```suggestion
                    err += esp_bt_mem_release(ESP_BT_MODE_BLE);
```

**Review by chshu — APPROVED** [review]

File: examples/platform/esp32/common/CommonDeviceCallbacks.cpp (hunk 1 above)
Comment (dhrishi) [reply examples/platform/esp32/common/CommonDeviceCallbacks.cpp:86]: This was the change done by the restyler (locally). When I tried to fix it as above, the restyler check in CI fails. See this: https://restyled.io/gh/project-chip/repos/connectedhomeip/jobs/3099457/patch Don't know how :-D

**Comment by dhrishi** [comment]
@bzbarsky-apple @andy31415 Can we merge this?

**Comment by bzbarsky-apple** [comment]
@dhrishi There are unresolved conversations.

**Comment by dhrishi** [comment]
> @dhrishi There are unresolved conversations.

I've resolved the same. It is the restyler that made the change. If undone, the CI fails as mentioned above

**Pull request merged**